### DISCUSSION TITLE
Recolour image with mapping

### DISCUSSION THREAD

**Discussion by Rhinemann** [ImageMagick/ImageMagick#7504] [Help] [answered]
### ImageMagick version

7.1.1-35

### Operating system, version and so on

OS: Kubuntu 23.10

### Description

I have an image that is 15% black in grayscale, I want to make it 15% this color: #eade8b, what would be the arguments for that?

**Comment by fmw42** [comment]
It would be better if you post an example image, but try this

`magick image.suffix -fuzz 0% -fill "#eade8b" -opaque "gray(15%)" result.suffix`

if your gray is truly 15% exactly.  If not increase the fuzz value to a few percent trying larger and larger values until you get all your gray to be the new color.

**Reply by Rhinemann** [reply]
Oh, okay, will post the image as soon as I'm able, and thanks will try the command once I can too.

**Reply by Rhinemann** [reply]
Also, looking at the command, wouldn't it just colour fill with my hex?
While what I want is the colour that's on the same position in a gradient between #ffffff and #eade8b as the gray-scale percentage.

So if they colour is in the middle between white and black it would become one in the middle between white and #eade8b.

**Reply by fmw42** [reply]
Sorry, I do not understand. It would be better to see your image and have you explain further where you want the color change.  The command I posted answered your question and will only color gray(15%) to your hex color.  But I think you want something else, but I am not sure exactly what that is, yet.

**Reply by Rhinemann** [reply]
So, here's what I mean.

![ForGold](https://github.com/user-attachments/assets/d946b709-fa2e-400e-90dd-2144955be7e1)
![For25%](https://github.com/user-attachments/assets/6193df06-3417-430a-bfd2-5543d45c257c)
![For50%](https://github.com/user-attachments/assets/eeb14986-1430-4feb-a1e2-2a4764a246b4)

I have these 3 images, they are grayscale, I wanted them to be on a scale from white to #eade8b like this:

![out100%](https://github.com/user-attachments/assets/4f5c0545-fa02-446b-bb37-12f9046f45b1)
![out25%](https://github.com/user-attachments/assets/dbbf34cd-9cd9-4b5d-88ee-0c6463edd114)
![out50%](https://github.com/user-attachments/assets/79365fd9-ce8f-4940-8f45-9bf06df85580)

**Comment by Rhinemann** [comment, marked as answer]
Found it, finally!
It's: `magick image.suffix  +level-colors "#eade8b",   result.suffix`
